### PR TITLE
Config/configurable doubtfire name

### PR DESCRIPTION
**Pull request by jakerenzella**
This PR will allow Doubtfire to configure the external name by editing the env.config.js file.

### CODE REVIEW

File: src/app/common/services/configuration-service.coffee [hunk 1]
```diff
@@ -0,0 +1,14 @@
+angular.module("doubtfire.common.services.configuration", [])
+
+.constant('external_name', '/* @echo EXTERNAL_NAME */')
```
Comment (alexcu) [inline src/app/common/services/configuration-service.coffee]: Double check the casing for angular constants -- should be camel

File: src/app/common/services/configuration-service.coffee [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Services for a configurable external name
+#
+.factory("configurationService", (external_name) ->
```
Comment (alexcu) [inline src/app/common/services/configuration-service.coffee]: No need for a service here 😊 We could just inject `externalName` anywhere we need to, like how we do with the `apiUrl`.

File: src/app/projects/project-portfolio-wizard/portfolio-review-step/portfolio-review-step.tpl.html [hunk 3]
```diff
@@ -20,13 +20,13 @@
   <div class="compile-scheduled text-center text-success" ng-show="portfolioIsCompiling">
     <i class="fa fa-check-circle fa-3x"></i>
     <h3>Compile Scheduled</h3>
-    <p>Your portfolio compilation is scheduled with Doubtfire. This process will take some time, check back soon to see if your portfolio is available.
+    <p>Your portfolio compilation is scheduled with {{ externalName  }}. This process will take some time, check back soon to see if your portfolio is available.
```
Comment (alexcu) [inline src/app/projects/project-portfolio-wizard/portfolio-review-step/portfolio-review-step.tpl.html]: Double spacing? Might just want to do a search and replace from `{{ externalName   }}` -> `{{externalName}}` (pretty sure angular docs don't space around the double braces).

File: src/app/sessions/states/sign-in/sign-in.coffee [hunk 4]
```diff
@@ -27,6 +29,9 @@ angular.module("doubtfire.sessions.states.sign-in", [])
 
   $scope.session = { remember_me: true }
 
+  # Get the confugurable, external name of Doubtfire
+  $scope.externalName = configurationService.getExternalName()
```
Comment (alexcu) [inline src/app/sessions/states/sign-in/sign-in.coffee]: Here we could just inject `externalName` constant and then expose to our `$scope`:

```
.controller("SignInCtrl", ($scope, $state, ..., externalName) ->
  ...
  $scope.externalName = externalName
)
```

File: build.config.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ module.exports = {
    */
   preprocess: {
     api: {
-      src: ['build/src/app/api/api-url.js'],
+      src: ['build/src/app/api/api-url.js','build/src/app/common/services/configuration-service.js'],
```
Comment (alexcu) [inline build.config.js]: Maybe separate `api` with `name` -- two different preprocesses.

File: env.config.js [hunk 6]
```diff
@@ -6,17 +6,20 @@ module.exports = {
 
     development: {
       API_URL  : 'http://localhost:3000/api',
-      DEV_MODE : 'true'
+      DEV_MODE : 'true',
```
Comment (alexcu) [inline env.config.js]: Align columns?

File: src/app/common/services/configuration-service.coffee (hunk 2 above)
Comment (alexcu) [inline src/app/common/services/configuration-service.coffee]: Delete the factory to remove the service and update according to `sign-in.coffee` example -- only need to inject the constant above.

File: src/app/common/services/configuration-service.coffee [hunk 7]
```diff
@@ -0,0 +1,14 @@
+angular.module("doubtfire.common.services.configuration", [])
```
Comment (alexcu) [inline src/app/common/services/configuration-service.coffee]: If we remove the configuration service, we could place this under a `config/external-name/external-name.coffee` file. Rename the module according to the folder structure (`doubtfire.config.external-name` in this case). You can see how we do the `contributors` constant under `contributors.coffee`.

File: src/app/projects/project-portfolio-wizard/portfolio-review-step/portfolio-review-step.tpl.html (hunk 3 above)
Comment (jakerenzella) [inline src/app/projects/project-portfolio-wizard/portfolio-review-step/portfolio-review-step.tpl.html]: ✅

File: src/app/common/services/configuration-service.coffee (hunk 7 above)
Comment (jakerenzella) [inline src/app/common/services/configuration-service.coffee]: ✅

File: src/app/common/services/configuration-service.coffee (hunk 1 above)
Comment (jakerenzella) [inline src/app/common/services/configuration-service.coffee]: ✅

File: src/app/sessions/states/sign-in/sign-in.coffee (hunk 4 above)
Comment (jakerenzella) [inline src/app/sessions/states/sign-in/sign-in.coffee]: ✅

**Comment by jakerenzella** [comment]
Hold up on merge sorry

**Comment by jakerenzella** [comment]
Ready to merge @alexcu

**Pull request merged**